framework/src/utils/MathUtils.C

```diff
@@ -14,181 +14,146 @@ namespace MathUtils
-    if (x < tol)
-      value = c1 * x * x * x + c2 * x * x + c3 * x + c4;
-    else
-      value = std::log(x);
-  }
-  else if (order == 1)
-  {
-    if (x < tol)
-      value = 3.0 * c1 * x * x + 2.0 * c2 * x + c3;
-    else
-      value = 1.0 / x;
-  }
-  else if (order == 2)
+  switch (derivative_order)
   {
-    if (x < tol)
-      value = 6.0 * c1 * x + 2.0 * c2;
-    else
-      value = -1.0 / (x * x);
-  }
-  else if (order == 3)
-  {
-    if (x < tol)
-      value = 6.0 * c1;
-    else
-      value = 2.0 / (x * x * x);
+    case 0:
+      return c1() * x * x + c2() * x + c3();
+
+    case 1:
+      return 2.0 * c1() * x + c2();
+
+    case 2:
+      return 2.0 * c1();
+
+    case 3:
+      return 0.0;
   }
-  return value;
+
+  mooseError("Unsupported derivative order ", derivative_order);
 }
 
 Real
-poly4Log(Real x, Real tol, int order)
+poly3Log(Real x, Real tol, unsigned int derivative_order)
 {
-  Real value = 0.0;
+  if (x >= tol)
+    return plainLog(x, derivative_order);
 
-  if (order == 0)
-  {
-    if (x < tol)
-      value = std::log(tol) + (x - tol) / tol - (x - tol) * (x - tol) / (2.0 * tol * tol) +
-              (x - tol) * (x - tol) * (x - tol) / (3.0 * tol * tol * tol) -
-              (x - tol) * (x - tol) * (x - tol) * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (5.0 * tol * tol * tol * tol * tol) -
-              (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = std::log(x);
-  }
-  else if (order == 1)
-  {
-    if (x < tol)
-      value = 1.0 / tol - 2.0 * (x - tol) / (2.0 * tol * tol) +
-              3.0 * (x - tol) * (x - tol) / (3.0 * tol * tol * tol) -
-              4.0 * (x - tol) * (x - tol) * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              5.0 * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (5.0 * tol * tol * tol * tol * tol) -
-              6.0 * (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = 1.0 / x;
-  }
-  else if (order == 2)
-  {
-    if (x < tol)
-      value = -2.0 * 1.0 / (2.0 * tol * tol) + 3.0 * 2.0 * (x - tol) / (3.0 * tol * tol * tol) -
-              4.0 * 3.0 * (x - tol) * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              5.0 * 4.0 * (x - tol) * (x - tol) * (x - tol) / (5.0 * tol * tol * tol * tol * tol) -
-              6.0 * 5.0 * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = -1.0 / (x * x);
-  }
-  else if (order == 3)
+  const auto c1 = [&]() { return 1.0 / (3.0 * tol * tol * tol); };
+  const auto c2 = [&]() { return -3.0 / (2.0 * tol * tol); };
+  const auto c3 = [&]() { return 3.0 / tol; };
+  const auto c4 = [&]() { return std::log(tol) - 11.0 / 6.0; };
```

Comment:
Accidentally placed a comment here instead of a review about why the lambdas. I understand now - they're only used once so this is similar enough to using a ternary and only executing them when needed.